backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/order/products_proposal/service/OrderLine.java

```diff
@@ -49,25 +50,31 @@ public class OrderLine
 
 	@NonNull
 	BigDecimal priceActual;
-	
+
 	@NonNull
 	BigDecimal priceEntered;
 
 	@NonNull
 	BigDecimal qtyEnteredCU;
 
+	@NonNull
+	AttributeSetInstanceId asiId;
+
 	int qtyEnteredTU;
 
 	String description;
 
 	boolean isMatching(
 			@NonNull final ProductId productId,
-			@Nullable final HUPIItemProductId packingMaterialId)
+			@Nullable final HUPIItemProductId packingMaterialId,
+			@NonNull Predicate<OrderLine> asiMatcher)
```

Comment:
@metas-rc @adi-stefan  IMHO makes no sense to have asiMatcher parameter here. You can simply call it from outside, no need to "call you to call me"